### ISSUE TITLE
Silent failure on too many color attachments

### ISSUE THREAD

**Issue by svenknobloch** [gfx-rs/gfx#2983]
GFX version: 0.3.0
Backend: Metal

When using the metal backend to try and run a rendering pass with more than 4 color attachments, the program panics with an array out of bounds exception when trying to call `begin_render_pass_inline` or the like. It is not very informative and after some digging, I found that the subpass description is silently dropping anything above the max amount of color attachments, via the `ArrayVec::collect` method. It would be very helpful to add an error message that tells the user that the max amount has been exceeded when the render pass object is created.

**Comment by kvark** [comment]
Agreed, a nicer error would be good. At the same time, validation is not in scope for the backend. It reports the maximum number of color attachments in the `Limits`, and if you don't obey those, you are on your own.
We'll accept a PR doing a better check, but this is largely out of scope (or very low priority).

You can get proper validation by running on Vulkan backend. What we could do, perhaps, is to provide a few [dev-sim](https://www.lunarg.com/new-vulkan-dev-sim-layer/) schemes, so that you test your app against limits of the non-Vulkan backends (while still using Vulkan validation).
